### PR TITLE
MovieDetailsPage

### PR DESCRIPTION
**Pull request by amyria3**
pls review :)

### CODE REVIEW

File: src/routes/MovieDetails.tsx [hunk 1]
```diff
@@ -1,20 +1,189 @@
...
-      <h1>{movie}</h1>
+function MovieDetails(): JSX.Element {
+  let { movieId } = useParams();
+
+  const navigate = useNavigate();
+  const [state, setState] = useState("start");
+
+  const currentData = useLoaderData() as CurrentData;
+
+  const movie_name: string = currentData.details.title;
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:61]: try using destructuring to create these bindings!
(optional)

File: src/routes/MovieDetails.tsx [hunk 2]
```diff
@@ -1,20 +1,189 @@
...
+  const movie_category: string = currentData.details.genres[0].name;
+  const movie_year: string = currentData.details.release_date;
+  const movie_score: number = currentData.details.popularity;
+  const movie_length: number | null = currentData.details.runtime;
+  const director: string = currentData.director;
+  const writer: string = currentData.writer;
+  const movie_synopsis = "";
+  const poster_path: string = posterUrl + currentData.details.poster_path;
+  console.log("LINE /! " + posterUrl + currentData.details.poster_path);
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:70]: we don't commit console.log to our main branches

File: src/routes/MovieDetails.tsx [hunk 3]
```diff
@@ -1,20 +1,189 @@
...
+
+  const MovieDetails = (
+    <div
+      id="page_wrapper"
+      className="bg-dark min-h-[667px] pb-[24px] py-[18px]"
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:75]: Please see if you can find an alternative for these square bracket noted utility classes, ideally use pb-6, etc... if 667px is a breakpoint that is used often, set a value for it in the tailwind config

File: src/routes/MovieDetails.tsx [hunk 4]
```diff
@@ -1,20 +1,189 @@
...
+    >
+      {/* <DetailHeaderLayout /> */}
+
+
+      <div id="content_wrapper" className="h-[568px] absolute top-[75px] mx-[18px]">
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:80]: see above

File: src/routes/MovieDetails.tsx [hunk 5]
```diff
@@ -1,20 +1,189 @@
...
+
+        <div id="image_wrapper"
+          className="mt-0 w-[100%] max-h-[211px] overflow-hidden rounded-lg"
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:83]: w-100% is the same as w-full, see above

File: src/routes/MovieDetails.tsx [hunk 6]
```diff
@@ -1,20 +1,189 @@
...
+                  </p>
+
+                  <p className="text-white-dimmed text-description">
+                    {movie_length}
+                  </p>
+                </div>
+
+                <p className="text-white-dimmed text-description">
+
+                  <span className="text-[rgba(34, 197, 94, 1)]">
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:119]: see above

File: src/routes/MovieDetails.tsx [hunk 7]
```diff
@@ -1,20 +1,189 @@
...
+                  </span>
+                </p>
+              </div>
+            </div>
+
+            <div id="cast_&_crew" className="pt-3 flex">
+
+              <div id="left" className="flex flex-col flex-1">
+
+                <div id="director" className="">
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:130]: empty className here

File: src/routes/oldCode.ts [hunk 8]
```diff
@@ -0,0 +1,119 @@
+// export type props = {
```
Comment (NoHara42) [inline src/routes/oldCode.ts:1]: we shouldn't commit unused files that serve no purpose to main

**Review by NoHara42 — CHANGES_REQUESTED** [review]
Some changes to be made, but looks good!

File: src/routes/oldCode.ts (hunk 8 above)
Comment (amyria3) [reply src/routes/oldCode.ts]: done

File: src/routes/MovieDetails.tsx (hunk 5 above)
Comment (amyria3) [reply src/routes/MovieDetails.tsx]: done

File: src/routes/MovieDetails.tsx (hunk 3 above)
Comment (amyria3) [reply src/routes/MovieDetails.tsx]: 667 is the height that dan have chosen for the prototype. 

I can put this and bg-color into home component in branch main OR i can define it for #root in tailwind theme. What should I do?

File: src/routes/MovieDetails.tsx (hunk 2 above)
Comment (amyria3) [reply src/routes/MovieDetails.tsx]: done

File: src/routes/MovieDetails.tsx (hunk 1 above)
Comment (amyria3) [reply src/routes/MovieDetails.tsx:63]: would love to but have no clue how :)

File: src/routes/MovieDetails.tsx (hunk 6 above)
Comment (amyria3) [reply src/routes/MovieDetails.tsx]: done

File: src/routes/MovieDetails.tsx (hunk 4 above)
Comment (dnmct) [reply src/routes/MovieDetails.tsx]: height should almost always be determined by the content + padding 
Ned is right

**Review by dnmct — CHANGES_REQUESTED** [review]

**Comment by amyria3** [comment]
page_wrapper has a height since height for all full-page components is not implemented yet :)

**Comment by amyria3** [comment]
I would like to link my commit d1d4ec3 to this branch, is this possible?

**Comment by amyria3** [comment]
ALL DONE juhuuu

File: src/routes/MovieDetails.tsx [hunk 9]
```diff
@@ -1,25 +1,209 @@
...
+
+    const director_name = people.crew.find(
+      (member) => member.job === "Director"
+    )?.name; //returned member, if 39 === true
+
+    const writer_name = people.crew.find(
+      (member) => member.job === "Writer"
+    )?.name;
+
+    console.log("TEST" + JSON.stringify(people));
```
Comment (franzwollang) [inline src/routes/MovieDetails.tsx:44]: Please remove console.log's before requesting a review

File: src/routes/MovieDetails.tsx [hunk 10]
```diff
@@ -1,25 +1,209 @@
...
     </div>
   );
+
+  return MovieDetails
 }
+
+export default MovieDetails
+
+
+//http://localhost:5173/315162 EXAMPLE URL TO TEST
```
Comment (franzwollang) [inline src/routes/MovieDetails.tsx:208]: please remove miscellaneous comments from code before requesting a review

**Review by franzwollang — COMMENTED** [review]
In general, looks good.

File: src/routes/MovieDetails.tsx (hunk 10 above)
Comment (amyria3) [reply src/routes/MovieDetails.tsx]: done

**Pull request merged**